File: src/main/java/cc/vileda/kalfor/Endpoint.java
package cc.vileda.kalfor;

import java.net.MalformedURLException;
import java.net.URL;
import java.util.Objects;


class Endpoint
{
		private final URL parsed;

		Endpoint(final String baseUrl) throws MalformedURLException
		{
				this.parsed = new URL(baseUrl);
		}

		Boolean isSSL()
		{
				return Objects.equals("https", scheme());
		}

		private String scheme()
		{
				return parsed.getProtocol();
		}

		String host()
		{
				return parsed.getHost();
		}

		int port()
		{
				if (parsed.getPort() == -1) { return parsed.getDefaultPort(); }
				else { return parsed.getPort(); }
		}
}

File: library/src/main/kotlin/cc/vileda/kalfor/handler/Endpoint.kt
package cc.vileda.kalfor.handler

import java.net.URL


class Endpoint(baseUrl: String) {
    private val parsed: URL

    val isSSL: Boolean?
        get() = "https" == scheme()

    fun scheme(): String {
        return parsed.protocol
    }

    fun host(): String {
        return parsed.host
    }

    fun port(): Int {
        if (parsed.port == -1) {
            return parsed.defaultPort
        } else {
            return parsed.port
        }
    }

    init {
        this.parsed = URL(baseUrl)
    }
}
